src/main/java/com/epam/learning/backendservices/springfoundation/config/SecurityConfig.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.learning.backendservices.springfoundation.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity;
+import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.web.server.SecurityWebFilterChain;
+
+@Configuration
+@EnableWebFluxSecurity

```

Comment:
It seems like you decided to make a classic Spring Web MVC app. You can't apply it.

Even though, you gave an access to anybody for actuator endpoints, I get 401 error.
This doesn't work - pathMatchers("/actuator/info/**", "/actuator/health/**", "/actuator/state/**").permitAll()

You to rewrite your app to be reactive or use this
```java
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.web.SecurityFilterChain;

registry.antMatchers("/actuator/info/**", "/actuator/health/**", "/actuator/state/**").permitAll()
```

```sh
export host=localhost:8080

curl -v ${host}/actuator/ # error 401 Unauthorized
curl -v ${host}/actuator/health
curl -v ${host}/actuator/info # error 401 Unauthorized
curl -v ${host}/actuator/state?id=123 # error 401 Unauthorized

export base64_password=$(echo -n 'dev:dev' | base64)

curl -v ${host}/actuator/ --header "Authorization: Basic ${base64_password}"
curl -v ${host}/actuator/health --header "Authorization: Basic ${base64_password}"
curl -v ${host}/actuator/info --header "Authorization: Basic ${base64_password}"
curl -v ${host}/actuator/state?id=123 --header "Authorization: Basic ${base64_password}"
```

Reply:
I am getting 403 error. I suggest applying the following:
1. since your default profile is set to DEV
```properties
spring.security.user.roles=DEVUSER
```
2. add this config
```java
@Bean
public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
    http.csrf().disable()
            .authorizeRequests()
            .antMatchers("/actuator/info", "/actuator/health").permitAll()
            .antMatchers("/actuator/state", "/user**").hasRole("DEVUSER")
            .anyRequest().authenticated();
    http.httpBasic();
    return http.build();
}
```